### PR TITLE
HOMS-179 Add ordering to Orders list

### PR DESCRIPTION
**Pull request by sugarfree1**

### CODE REVIEW

File: app/filters/list_orders_filter.rb [hunk 1]
```diff
@@ -83,6 +83,33 @@ def filter_with_dispatch(rel, column, value)
...
+        'user'                => 'user_id',
+        'ext_code'            => 'ext_code',
+        'archived'            => 'archived',
+        'estimated_exec_date' => 'estimated_exec_date'
+    }
+
+    def sort_by(rel, column, order)
+      if column
+        if MAPPING.keys.include?(column)
+          rel.reorder("#{MAPPING[column]} #{order}").order('created_at DESC')
```
Comment (TimAle) [inline app/filters/list_orders_filter.rb]: Maybe we can add `nulls last`?

**Comment by sugarfree1** [comment]
Updated

File: app/filters/list_orders_filter.rb [hunk 2]
```diff
@@ -83,6 +83,37 @@ def filter_with_dispatch(rel, column, value)
...
+        'archived'            => %w(archived),
+        'estimated_exec_date' => %w(estimated_exec_date)
+    }
+
+    def sort_by(rel, column, order)
+      if column
+        if MAPPING.keys.include?(column)
+          rel.includes(:order_type).includes(:user).reorder(render_order_by_clause(MAPPING[column], order)).order('orders.created_at DESC')
+        else
+          rel.reorder("data->>'#{column}' #{order}").order('orders.created_at DESC')
```
Comment (TimAle) [inline app/filters/list_orders_filter.rb]: `nulls last`

File: spec/features/orders/list_orders_spec.rb [hunk 3]
```diff
@@ -292,4 +319,56 @@
       expect(order_list_table_cols).to include('Code')
     end
   end
+
+  feature 'with ordering' do
+    let!(:support_request_order_for_ordering)  { FactoryGirl.create(:order,
```
Comment (TimAle) [inline spec/features/orders/list_orders_spec.rb]: `)  {` two spaces.

File: spec/features/orders/list_orders_spec.rb [hunk 4]
```diff
@@ -292,4 +319,56 @@
...
+      # "Order type"
+      change_select2_value('order_type_id', 'Support request')
+      search_button.click
+      wait_for_ajax
+      expect(orders_list).to eq([[], order_3, order_2])
+
+      # Add Contract number to orders table
+      custom_fields_button.click
+      custom_fields_list('Contract number').click
+      custom_fields_button.trigger('click')
```
Comment (TimAle) [inline spec/features/orders/list_orders_spec.rb]: But we already have `click_on_multiselect_options(id, options)`, why didn't you use it?

**Comment by sugarfree1** [comment]
Updated

**Pull request merged**